substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ResourceConfigurationParser.java

```diff
@@ -136,12 +108,106 @@ private static Locale parseLocale(Object input) {
+        String tmpGlob = glob + '#';
+        StringBuilder sb = new StringBuilder();
+
+        int quoteStartIndex = 0;
+        Wildcard previousWildcard = Wildcard.START;
+        for (int i = 0; i < tmpGlob.length(); i++) {
+            char c = tmpGlob.charAt(i);
+            Wildcard currentWildcard = previousWildcard.next(c);
+
+            boolean wildcardStart = previousWildcard == Wildcard.START && currentWildcard != Wildcard.START;
+            if (wildcardStart && quoteStartIndex != i) {
+                /* start of the new wildcard => quote previous content */
+                sb.append(Pattern.quote(tmpGlob.substring(quoteStartIndex, i)));
+            }
+
+            boolean consecutiveWildcards = previousWildcard == Wildcard.DOUBLE_STAR_SLASH && currentWildcard != Wildcard.START;
+            boolean wildcardEnd = previousWildcard != Wildcard.START && currentWildcard == Wildcard.START;
+            if (wildcardEnd || consecutiveWildcards) {
+                /* end of the wildcard => append regex and move start of next quote after it */
+                sb.append(previousWildcard.regex);
+                quoteStartIndex = i;
+            }
+            previousWildcard = currentWildcard;
+        }
+        /* remove the last char we added artificially */
+        tmpGlob = tmpGlob.substring(0, tmpGlob.length() - 1);
+        if (quoteStartIndex < tmpGlob.length()) {
+            sb.append(Pattern.quote(tmpGlob.substring(quoteStartIndex)));
+        }
+        return sb.toString();
+    }
+
+    /**
+     * This enum acts like a state machine that helps to identify glob wildcards.
+     */
+    private enum Wildcard {
+        START("") {
+            @Override
+            public Wildcard next(char c) {
+                return c == '*' ? STAR : START;
+            }
+        },
+        STAR("[^/]*") {
+            @Override
+            public Wildcard next(char c) {
+                return c == '*' ? DOUBLE_STAR : START;
+            }
+        },
+        DOUBLE_STAR(".*") {
+            @Override
+            public Wildcard next(char c) {
+                return c == '/' ? DOUBLE_STAR_SLASH : START;
+            }
+        },
+        DOUBLE_STAR_SLASH("([^/]*(/|$))*") {
+            @Override
+            public Wildcard next(char c) {
+                return c == '*' ? STAR : START;
+            }
+        };
+
+        final String regex;
+
+        Wildcard(String val) {
+            regex = val;
+        }
+
+        public abstract Wildcard next(char c);
+    }
+
+    protected void parseGlobsObject(Object globsObject) {
+        List<Object> globs = asList(globsObject, "Attribute 'globs' must be a list of glob patterns");
+        for (Object object : globs) {
+            parseGlobEntry(object, (condition, module, resource) -> registry.addResources(condition, globToRegex(module, resource)));
+        }
+    }
+
+    private interface GlobPatternConsumer<T> {
+        void accept(T a, String b, String c);
+    }
```

Comment:
This code is new (instead of backported more code from https://github.com/oracle/graal/pull/8715) in order to reduce the touched classes. Although having less code touched is preferred, this feels quite risky to me as it will only be "tested" in the LTS branches (and probably not to a great extend). 

@graalvm/community-maintainers @simonis WDYT? How do you handle similar cases in OpenJDK backports?

Reply:
I don't think such a feature would ever be downported in the OpenJDK in the first place :)

In the end it is always a judgement call between the ease of a downport (this one is already quite big) and long term maintainability. I currently can't oversee which existing functionality in the upstream version this new code is replacing and how many additional changes would need to be downorted in oder to make this code look more like the upstream version. If you could detail the alternative (i.e. the additional changes that would need to be downported), that would be helpful.

In general, I have problems to understand the reason behind this downport (see my general comment below). For me, the only justification to accept such a downport would be to maintain compatibility with the corresponding Oracle GraalVM verision 23.1. But in that case we wouldn't have a big choice anyway and should probably accept the exact change that was downported to Oracle GraalVM 23.1 (although we can't see it).

Reply:
See https://github.com/graalvm/graalvm-community-jdk21u/pull/23#discussion_r1776713643 for the reasoning

Reply:
> I don't think such a feature would ever be downported in the OpenJDK in the first place :)

I see :)

> If you could detail the alternative (i.e. the additional changes that would need to be downported), that would be helpful.

For the sake of simplicity (and ease of maintenance), let's say that we would need to backport https://github.com/oracle/graal/pull/8715 as a whole (this would also address my concern in https://github.com/graalvm/graalvm-community-jdk21u/pull/23#discussion_r1775277293)

> In general, I have problems to understand the reason behind this downport (see my general comment below). 

Let's keep this discussion in the general comment.